### PR TITLE
MACS2: Added option "--SPMR" to save signal per million reads

### PR DESCRIPTION
**Pull request by sveinugu**
--SPMR is needed to generate ChIP-seq signal tracks according to the MACS2 documentation:

https://github.com/taoliu/MACS/wiki/Build-Signal-Track

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

**Comment by sveinugu** [comment]
All tests pass locally.

**Comment by nsoranzo** [comment]
Thanks @sveinugu, I fixed `macs2_predictd`.

File: tools/macs2/macs2_callpeak.xml [hunk 1]
```diff
@@ -219,6 +220,9 @@
                     help="By default, the bigger dataset will be scaled down towards the smaller dataset, which will lead to smaller p/qvalues and more specific results. Keep in mind that scaling down will bring down background noise more. (--to-large). Default: No"/>
                 <param name="nolambda" type="boolean" truevalue="--nolambda" falsevalue="" checked="False" optional="True"
                     label="Use fixed background lambda as local lambda for every peak region" help="up to 9X more time consuming (--nolambda). Default: No"/>
+                <param name="spmr" type="boolean" truevalue="--SPMR" falsevalue="" checked="False" optional="True"
```
Comment (nsoranzo) [inline tools/macs2/macs2_callpeak.xml]: I don't think that `optional="True"` makes sense for a boolean param.

File: tools/macs2/macs2_callpeak.xml [hunk 2]
```diff
@@ -219,6 +220,9 @@
...
+                    label="Save signal per million reads for fragment pileup profiles"
+                    help="Requires 'Scores in bedGraph files (--bdg)' output to be selected. (--SPMR). Default: No"/>
```
Comment (nsoranzo) [inline tools/macs2/macs2_callpeak.xml]: Can you please remove `(--SPMR)` from here and add the attribute `argument="--SPMR"` between `name` and `type`? See https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#parameter-name-argument-and-help

**Comment by nsoranzo** [comment]
https://github.com/galaxyproject/tools-iuc/pull/2138

**Pull request merged**